core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -602,10 +606,14 @@ private DefaultFunctionResolver utc_timestamp() {
    */
   private DefaultFunctionResolver week(BuiltinFunctionName week) {
     return define(week.getName(),
+        implWithProperties((functionProperties, arg) -> DateTimeFunction.weekOfYearToday(new ExprIntegerValue(0),
+            functionProperties.getQueryStartClock()), INTEGER, TIME),
         impl(nullMissingHandling(DateTimeFunction::exprWeekWithoutMode), INTEGER, DATE),
         impl(nullMissingHandling(DateTimeFunction::exprWeekWithoutMode), INTEGER, DATETIME),
         impl(nullMissingHandling(DateTimeFunction::exprWeekWithoutMode), INTEGER, TIMESTAMP),
         impl(nullMissingHandling(DateTimeFunction::exprWeekWithoutMode), INTEGER, STRING),
+        implWithProperties((functionProperties, arg1, arg2) -> DateTimeFunction.weekOfYearToday(arg2,
```

Comment:
you should be able to replace `arg2` with `mode` or `modeArg`

Reply:
It can be used in other places in the file, such as in the function `exprWeekWithoutMode`

Reply:
Fixed in 85c93e677ba64de401fa4b4115acfe428005e003